tmt/steps/execute/internal.py

```diff
@@ -38,32 +42,40 @@ class ExecuteInternal(tmt.steps.execute.ExecutePlugin):
 
     _data_class = ExecuteInternalData
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: Any, **kwargs: Any):
         super().__init__(*args, **kwargs)
         self._previous_progress_message = ""
         self.scripts = SCRIPTS
 
     @classmethod
-    def options(cls, how=None):
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
         """ Prepare command line options for given method """
-        options = []
-        # Shell script as a test
-        options.append(click.option(
-            '-s', '--script', metavar='SCRIPT', multiple=True,
-            help='Shell script to be executed as a test.'))
-        # Interactive mode
-        options.append(click.option(
-            '-i', '--interactive', is_flag=True,
-            help='Run in interactive mode, do not capture output.'))
-        # Disable interactive progress bar
-        options.append(click.option(
-            '--no-progress-bar', is_flag=True,
-            help='Disable interactive progress bar showing the current test.'))
-        return options + super().options(how)
+        options = super().options(how)
+        options[:0] = [
+            click.option(
+                '-s', '--script', metavar='SCRIPT', multiple=True,
+                help='Shell script to be executed as a test.'),
+            # Interactive mode
+            click.option(
+                '-i', '--interactive', is_flag=True,
+                help='Run in interactive mode, do not capture output.'),
+            # Disable interactive progress bar
+            click.option(
+                '--no-progress-bar', is_flag=True,
+                help='Disable interactive progress bar showing the current test.')
+            ]
+        return options
+
+    def wake(self) -> None:
+        """ Wake up the plugin, process data, apply options """
+        super().wake()
+        # Make sure that script is a list
+        tmt.utils.listify(cast(Dict[Any, Any], self.data), keys=['script'])
```

Comment:
Not needed, ‘data.script’ is already normalized.

Reply:
Well, `data.script` may be, but the problem is passing the first parameter `self.data` which is of type `StepData`, but `listify` expecting `data: Union[Tuple[Any, ...], List[Any], str, Dict[Any, Any]]`, resulting still in:

```
tmt/steps/execute/internal.py:73: error: Argument 1 to "listify" has incompatible type "StepData"; expected "Union[Tuple[Any, ...], List[Any], str, Dict[Any, Any]]"  [arg-type]
```

Reply:
Sorry, I mean the whole `listify` call - `self.data.script` should be already normalized, i.e. a list, thanks to `_normalize_script = tmt.utils.NormalizeKeysMixin._normalize_string_list` in step data class.

Reply:
I see, removed.